### ISSUE TITLE
Remove min/max in computing ByteCellType for GDALDataset

### ISSUE THREAD

**Issue by echeipesh** [locationtech/geotrellis#3219]
Related: https://github.com/locationtech/geotrellis/pull/3150
Requires: https://github.com/geotrellis/gdal-warp-bindings/issues/71

This is a performance change that avoids finding min/max values for byte rasters when reading them with `GDALRasterSource`.

However, `GDALRasterSource` and `GeoTiffRasterSource` should have identical behavior.
The logic should be consistent with  `TiffTags` parsing:
https://github.com/locationtech/geotrellis/blob/92b4da15da80b4ee2d2ce0bb1936a8a9a5a3b356/raster/src/main/scala/geotrellis/raster/io/geotiff/tags/TiffTags.scala#L279-L291

**Comment by pomadchin** [comment]
Closing it as won't fix; also in terms of this task I discovered a slight CellTypes inconsistency between GeoTiffRasterSource and GDALRasterSource, see: https://github.com/locationtech/geotrellis/issues/3232

**Comment by pomadchin** [comment]
Replaced with https://github.com/locationtech/geotrellis/issues/3232
